test/e2e/garbagecollector/garbagecollector_test.go

```diff
@@ -0,0 +1,642 @@
+
+	cfg := server.BaseConfig(t)
+
+	kcpClusterClient, err := kcpclient.NewForConfig(cfg)
+	require.NoError(t, err, "error creating kcp cluster client")
+
+	clusterCfg := kcpclienthelper.SetCluster(rest.CopyConfig(cfg), apiProviderClusterName)
+	apiProviderClient, err := kcpclient.NewForConfig(clusterCfg)
+	require.NoError(t, err)
+
+	dynamicClusterClient, err := kcpdynamic.NewForConfig(cfg)
+	require.NoError(t, err, "failed to construct dynamic cluster client for server")
+
+	t.Logf("Create the cowboy APIResourceSchema")
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(apiProviderClient.Discovery()))
+	err = helpers.CreateResourceFromFS(ctx, dynamicClusterClient.Cluster(apiProviderClusterName), mapper, nil, "apiresourceschema_cowboys.yaml", testFiles)
+	require.NoError(t, err)
+
+	t.Logf("Create an APIExport for it")
+	cowboysAPIExport := &apisv1alpha1.APIExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "today-cowboys",
+		},
+		Spec: apisv1alpha1.APIExportSpec{
+			LatestResourceSchemas: []string{"today.cowboys.wildwest.dev"},
+		},
+	}
+	_, err = kcpClusterClient.ApisV1alpha1().APIExports().Create(logicalcluster.WithCluster(ctx, apiProviderClusterName), cowboysAPIExport, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	// Test multiple workspaces in parallel
+	for i := 0; i < 3; i++ {
+		i := i
+		t.Run(fmt.Sprintf("tc%d", i), func(t *testing.T) {
+			t.Parallel()
+
+			c, cancelFunc := context.WithCancel(ctx)
+			t.Cleanup(cancelFunc)
+
+			userClusterName := framework.NewWorkspaceFixture(t, server, orgClusterName, framework.WithName("gc-api-binding-%d", i))
+
+			t.Logf("Create a binding in the user workspace")
+			binding := &apisv1alpha1.APIBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cowboys",
+				},
+				Spec: apisv1alpha1.APIBindingSpec{
+					Reference: apisv1alpha1.ExportReference{
+						Workspace: &apisv1alpha1.WorkspaceExportReference{
+							Path:       apiProviderClusterName.String(),
+							ExportName: cowboysAPIExport.Name,
+						},
+					},
+				},
+			}
+
+			kubeClusterClient, err := kcpkubernetesclientset.NewForConfig(cfg)
+			require.NoError(t, err, "error creating kube cluster client")
+
+			kcpClusterClient, err := kcpclient.NewForConfig(cfg)
+			require.NoError(t, err, "error creating kcp cluster client")
+
+			_, err = kcpClusterClient.ApisV1alpha1().APIBindings().Create(logicalcluster.WithCluster(c, userClusterName), binding, metav1.CreateOptions{})
+			require.NoError(t, err)
+
+			t.Logf("Wait for the binding to be ready")
+			framework.Eventually(t, func() (bool, string) {
+				binding, err := kcpClusterClient.ApisV1alpha1().APIBindings().Get(logicalcluster.WithCluster(c, userClusterName), binding.Name, metav1.GetOptions{})
+				require.NoError(t, err, "error getting binding %s", binding.Name)
+				return conditions.IsTrue(binding, apisv1alpha1.InitialBindingCompleted), fmt.Sprintf("binding not bound: %s", toYaml(binding))
```

Comment:
I know you just carried this forward from other tests, but let's not use `toYaml()`: https://github.com/kcp-dev/kcp/pull/2273